src/code/z_onepointdemo.c

```diff
@@ -1,55 +1,1365 @@
+        case ACTORCAT_PLAYER:
+        case ACTORCAT_PROP:
+        case ACTORCAT_DOOR:
+            timer = 30;
+            break;
+        case ACTORCAT_NPC:
+        case ACTORCAT_ITEMACTION:
+        case ACTORCAT_CHEST:
+            timer = 100;
+            break;
+        case ACTORCAT_EXPLOSIVE:
+        case ACTORCAT_ENEMY:
+        case ACTORCAT_MISC:
+        case ACTORCAT_BOSS:
+        default:
+            osSyncPrintf(VT_COL(YELLOW, BLACK) "actor attention demo camera: %d: unkown part of actor %d\n" VT_RST,
+                         globalCtx->state.frames, actor->category);
+            timer = 30;
+            break;
+    }
+    osSyncPrintf(VT_FGCOL(CYAN) "%06u:" VT_RST " actor attention demo camera: request %d ", globalCtx->state.frames,
+                 actor->category);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_onepointdemo/func_800806BC.s")
+    // If the previous attention demo has the same category, skip this actor.
+    if (actor->category == vLastHigherCat) {
+        osSyncPrintf("→ " VT_FGCOL(PURPLE) "×" VT_RST " (%d)\n", actor->id);
+        return SUBCAM_NONE;
+    }
+    osSyncPrintf("→ " VT_FGCOL(BLUE) "○" VT_RST " (%d)\n", actor->id);
+    vDemoCamIdx = OnePointDemo_Init(globalCtx, 5010, timer, actor, vParentCamIdx);
+    if (vDemoCamIdx == SUBCAM_NONE) {
+        osSyncPrintf(VT_COL(RED, WHITE) "actor attention demo: give up! \n" VT_RST, actor->id);
+        return SUBCAM_NONE;
+    } else {
+        s32* data = (s32*)&globalCtx->cameraPtrs[vDemoCamIdx]->data1;
 
-void func_8008070C() {
-    D_80120130 = 0;
+        *data = NA_SE_SY_CORRECT_CHIME;
+        return vDemoCamIdx;
+    }
 }
 
-void func_80080718() {
-    D_80120130 = 1;
+// Adds an attention demo to the list with the specified sound effect
+s32 OnePointDemo_AttentionSetSfx(GlobalContext* globalCtx, Actor* actor, s32 sfxId) {
+    s32 demoCamIdx = OnePointDemo_Attention(globalCtx, actor);
+
+    if (demoCamIdx != SUBCAM_NONE) {
+        s32* data = (s32*)&globalCtx->cameraPtrs[demoCamIdx]->data1;
+
+        *data = sfxId;
+    }
+    return demoCamIdx;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_onepointdemo/func_80080728.s")
+// unused
+void OnePointDemo_EnableAttention() {
+    sDisableAttention = false;
+}
+
+// unused
+void OnePointDemo_DisableAttention() {
+    sDisableAttention = true;
+}
+
+s32 OnePointDemo_CheckForCategory(GlobalContext* globalCtx, s32 category) {
+    Camera* parentCam = globalCtx->cameraPtrs[MAIN_CAM];
+
+    while (parentCam->childCamIdx != SUBCAM_FREE) {
+        parentCam = globalCtx->cameraPtrs[parentCam->childCamIdx];
+        if ((parentCam == NULL) || (parentCam->setting != CAM_SET_DEMO4)) {
+            break;
+        } else if (category == parentCam->target->category) {
+            return true;
+        }
+    }
+    return false;
+}
 
-void func_80080788(UNK_TYPE arg0, UNK_TYPE arg1) {
+// unused, also empty.
+void OnePointDemo_Noop(GlobalContext* globalCtx, UNK_TYPE arg1) {
```

Comment:
id say personally we can use s32 for that unk arg, we're never gonna find out what it is.
I treat the UNK types as "filler for now and will properly replace later"